include/glow/Base/Tensor.h

```diff
@@ -682,9 +695,11 @@ template <class ElemTy> class Handle final {
   const_iterator begin() const { return tensor_->getRawDataPointer<ElemTy>(); }
 
   /// \returns an iterator referring to the past-the-end element.
-  iterator end() { return tensor_->getRawDataPointer<ElemTy>() + size(); }
+  iterator end() {
+    return tensor_->getRawDataPointer<ElemTy>() + tensor_->actualSize();
```

Comment:
Don't you think it will break iteration? Because iterator now can point to the "garbage" elements after incrementation beyond actual dimension size.

Reply:
Yes, it will break iteration. Now we need to be extra careful about where and why we use iteration.

It is worth to mention that I don't have intention to support custom-aligned Tensors for every backend. It is OK if Interpreter doesn't work correctly for non-1-aligned Tensors. The goal of this PR is to simply allow having such Tensors.

Reply:
> Yes, it will break iteration. Now we need to be extra careful about where and why we use iteration.

Maybe we should make a custom iterator?

Reply:
Implemented custom iterator in a separate commit

Reply:
May be you can add assert() to the iterator begin(), end() methods, for non-zero strides? It will prevent the incorrect usage, till you will implement iterators with strides.

Reply:
Yesterday I added a commit to this PR, which implements `HandleIterator` class.